### PR TITLE
Equality testing instead of identity

### PR DESCRIPTION
**Pull request by dfalbel**
Identity comparing here will result in errors when the input string is created in non trivial ways. for example:

```
import tensorflow as tf
import tensorflow.keras as keras

standardize = ''.join(['lower', '_and_strip_punctuation'])
layer = keras.layers.experimental.preprocessing.TextVectorization(standardize=standardize)
layer(tf.constant([['hello'], ['hello world']]))

# ValueError: lower_and_strip_punctuation is not a supported standardization. TextVectorization supports the following options for `standardize`: None, 'lower_and_strip_punctuation', or a Callable.

split = ''.join('whitespace')
layer = keras.layers.experimental.preprocessing.TextVectorization(split=split)
layer(tf.constant([['hello'], ['hello world']]))

# ValueError: whitespace is not a supported splitting.TextVectorization supports the following options for `split`: None, 'whitespace', or a Callable.
```

This was tested with 

```
>>> tf.__version__
'2.1.0-dev20191119'
```

And Python3

### CODE REVIEW

**Comment by dfalbel** [comment]
@googlebot I signed it!

**Comment by mihaimaruseac** [comment]
Looks good to me, but can you also write a test please? The one example you mentioned at the start of the issue should be good.

**Comment by dfalbel** [comment]
@mihaimaruseac just added the tests.

**Review by mihaimaruseac — DISMISSED** [review]
Make sure pylint also passes on this.

I'm approving, there will be tests running and we can see the progress there

**Comment by mihaimaruseac** [comment]
```
FAIL: Found 2 non-whitelisted pylint errors:
tensorflow/python/keras/layers/preprocessing/text_vectorization_test.py:475: [C0326(bad-whitespace), ] Exactly one space required after comma

tensorflow/python/keras/layers/preprocessing/text_vectorization_test.py:485: [C0326(bad-whitespace), ] Exactly one space required after comma
```

**Comment by dfalbel** [comment]
@mihaimaruseac Fixed the lining issues. I'm new to Python, running:

```
pylint --rcfile=/tmp/pylintrc tensorflow/python/keras/layers/preprocessing/text_vectorization_test.py
```
 still returns a lot of warnings but they don't seem related to this PR.

**Review by mihaimaruseac — DISMISSED** [review]

File: tensorflow/python/keras/layers/preprocessing/text_vectorization_test.py [hunk 1]
```diff
@@ -470,6 +470,25 @@ def test_splitting_with_invalid_split_arg(self):
     with self.assertRaisesRegex(ValueError, ".*is not a supported splitting.*"):
       _ = layer(input_data)
 
+  def test_standardize_with_no_identical_argument(self):
+    input_array = np.array([["hello world"]])
+    expected_output = np.array([[1, 1]])
+
+    standardize = "".join(["lower", "_and_strip_punctuation"])
+    layer = get_layer_class()(standardize=standardize)
+    output = layer(input_array).numpy()
```
Comment (mihaimaruseac) [inline tensorflow/python/keras/layers/preprocessing/text_vectorization_test.py]: This fails (internally) with

```
    output = layer(input_array).numpy()
AttributeError: 'Tensor' object has no attribute 'numpy'
```

**Comment by gbaned** [comment]
@dfalbel Could you please check reviewer comments and keep us posted. Thanks!

**Comment by dfalbel** [comment]
yes, I'll do it

**Review by mihaimaruseac — APPROVED** [review]

**Pull request merged**